src/audio/multiband_drc/multiband_drc.c

```diff
@@ -126,6 +186,67 @@ static int multiband_drc_init_coef(struct processing_module *mod, int16_t nch, u
 	comp_info(dev, "multiband_drc_init_coef(), initializing %i-way crossover",
 		  config->num_bands);
 
+	/* Calculate total size for the contiguous memory block */
+	size_t total_size = 0;
+
+	for (ch = 0; ch < nch; ++ch) {
+		size_t emp_deemp_size = sizeof(struct sof_eq_iir_biquad) * SOF_EMP_DEEMP_BIQUADS;
+		size_t crossover_delay_size = sizeof(uint64_t) * CROSSOVER_NUM_DELAYS_LR4;
+
+		/*  For first block (emphasis)*/
+		total_size += emp_deemp_size + crossover_delay_size;
+		/*  For second block (crossover delays)*/
+		total_size += emp_deemp_size + crossover_delay_size;
+		/*  For third block (deemphasis)*/
+		total_size += emp_deemp_size + crossover_delay_size;
+
+		/* Each block needs to be properly aligned */
+		total_size += (emp_deemp_size % __alignof__(struct sof_eq_iir_biquad) != 0) ?
+			__alignof__(struct sof_eq_iir_biquad) -
+			(emp_deemp_size % __alignof__(struct sof_eq_iir_biquad)) : 0;
+
+		total_size += (crossover_delay_size % __alignof__(uint64_t) != 0) ?
+			__alignof__(uint64_t) -
+			(crossover_delay_size % __alignof__(uint64_t)) : 0;
+	}
+
+	/* For DRC states */
+	total_size += sizeof(struct drc_state) * num_bands * nch;
+
+	/* Define offsets and pointer lists for allocations */
+	void *ptrs[3 * nch * 3 + num_bands];
+	size_t sizes[3 * nch * 3 + num_bands];
+
+	for (ch = 0; ch < nch; ++ch) {
+		size_t emp_deemp_size = sizeof(struct sof_eq_iir_biquad) * SOF_EMP_DEEMP_BIQUADS;
+		size_t crossover_delay_size = sizeof(uint64_t) * CROSSOVER_NUM_DELAYS_LR4;
+		int base = ch * 3;
+
+		/*  Align and set sizes for first block (emphasis/deemphasis) */
+		sizes[base] = emp_deemp_size;
+		sizes[base + 1] = emp_deemp_size;
+		sizes[base + 2] = crossover_delay_size;
+
+		/* Align and set sizes for second block (crossover delays) */
+		sizes[3 * nch + base] = emp_deemp_size;
+		sizes[3 * nch + base + 1] = emp_deemp_size;
+		sizes[3 * nch + base + 2] = crossover_delay_size;
+
+		/* Align and set sizes for third block (emphasis/deemphasis) */
+		sizes[2 * 3 * nch + base] = emp_deemp_size;
+		sizes[2 * 3 * nch + base + 1] = emp_deemp_size;
+		sizes[2 * 3 * nch + base + 2] = crossover_delay_size;
+	}
+	/* Set sizes for DRC state blocks */
+	for (i = 0; i < num_bands; ++i)
+		sizes[3 * nch + i] = sizeof(struct drc_state);
+
+	void *base_addr = allocate_contiguous_memory(dev, total_size,
+						     ptrs, sizes, 3 * nch + num_bands);
+
+	if (!base_addr)
```

Comment:
I don't see any modifications to the free logic in this commit which has me really concerned this is creating a massive double free